index.d.ts

```diff
@@ -43,7 +43,7 @@ export interface Action {
  *
  * @template S State object type.
  */
-export type Reducer<S> = <A extends Action>(state: S | undefined, action: A) => S;
+export type Reducer<S, A extends Action = Action> = (state: S, action: A) => S;
```

Comment:
Redux itself passes `undefined` to the reducer on Store init. Typings must reflect this.